src/device_trezor/trezor/protocol.cpp

```diff
@@ -976,6 +991,41 @@ namespace tx {
     m_ct.enc_salt1 = ack->salt();
     m_ct.enc_salt2 = ack->rand_mult();
     m_ct.enc_keys = ack->tx_enc_keys();
+
+    // Opening the sealed signatures
+    if (client_version() >= 3){
+      if(!ack->has_opening_key()){
+        throw exc::ProtocolException("Client version 3+ requires sealed signatures");
+      }
+
+      for(size_t i = 0; i < m_ct.signatures.size(); ++i){
+        std::string nonce = compute_sealing_key(ack->opening_key(), i, true);
+        std::string key = compute_sealing_key(ack->opening_key(), i, false);
+        size_t plen = m_ct.signatures[i].size() - crypto::chacha::TAG_SIZE;
```

Comment:
Is this checked somewhere so that can now wrap down to positive ?

Reply:
Good point! I've added the check to be sure.